### PR TITLE
Add --insecure var for allowing insecure ssl

### PR DESCRIPTION
**Pull request by TheLastProject**
Useful for, for example, localhost checks over SSL

### CODE REVIEW

File: check_http_expect [hunk 1]
```diff
@@ -153,11 +154,13 @@ while [ $# -gt 0  ]; do
 		# Final target url
 		--url)
 			url="$1"
+			shift
```
Comment (cytopia) [inline check_http_expect:157]: `shift` is already happening here at the top of the loop: https://github.com/cytopia/check_http_expect/pull/2/files#diff-dd5699a63f65e4bc94d8545accf83734R151

The current implementation will break parameter parsing.

Try out:
```
$ ./check_http_expect --url http://google.de

Error, You must specify what you are looking for.
Usage:  check_http_expect --url <url> --find <string> [--insecure] [--huser <user>] [--hpass <pass>] [--lurl <url>] [--cookie <data> [--cookie <data>]][--ldata <data> [--ldata <data>]]
OR      check_http_expect --check
OR      check_http_expect --help
OR      check_http_expect --version
```

File: check_http_expect [hunk 2]
```diff
@@ -153,11 +154,13 @@ while [ $# -gt 0  ]; do
...
 			;;
 
 		# Post parameter
 		--lurl)
 			lurl="$1"
+			shift
```
Comment (cytopia) [inline check_http_expect:163]: Same as above

File: check_http_expect [hunk 3]
```diff
@@ -166,19 +169,23 @@ while [ $# -gt 0  ]; do
 			else
 				ldata="${ldata} --data \"$1\""
 			fi
+			shift
```
Comment (cytopia) [inline check_http_expect:172]: Same as above

File: check_http_expect [hunk 4]
```diff
@@ -166,19 +169,23 @@ while [ $# -gt 0  ]; do
...
 			;;
 
 		# Htaccess parameters
 		--huser)
 			huser="$1"
+			shift
```
Comment (cytopia) [inline check_http_expect:178]: Same as above

File: check_http_expect [hunk 5]
```diff
@@ -166,19 +169,23 @@ while [ $# -gt 0  ]; do
...
 			;;
 		--hpass)
 			hpass="$1"
+			shift
```
Comment (cytopia) [inline check_http_expect:182]: Same as above

File: check_http_expect [hunk 6]
```diff
@@ -166,19 +169,23 @@ while [ $# -gt 0  ]; do
...
 			;;
 
 		# String to look for on final url
 		--find)
 			find="$1"
+			shift
```
Comment (cytopia) [inline check_http_expect:188]: Same as above

File: check_http_expect [hunk 7]
```diff
@@ -187,14 +194,18 @@ while [ $# -gt 0  ]; do
 			else
 				cookies="${cookies} --cookie \"$1\""
 			fi
+			shift
```
Comment (cytopia) [inline check_http_expect:197]: Same as above

File: check_http_expect [hunk 8]
```diff
@@ -187,14 +194,18 @@ while [ $# -gt 0  ]; do
...
+			;;
+		# Insecure
+		--insecure)
```
Comment (cytopia) [inline check_http_expect:200]: :+1:

**Review by cytopia — CHANGES_REQUESTED** [review]
@TheLastProject thanks for your PR adding the `--insecure` flag. There's however one additional change made that breaks the parameter parsing. Can you please address the  requested changes.

File: check_http_expect (hunk 1 above)
Comment (TheLastProject) [reply check_http_expect:157]: I removed the shift at the bottom to compensate for this, so we can have a flag that doesn't need a parameter. Are you sure the parsing is broken? That error message to me implies you forgot to set --find which is a required flag as well.

Comment (cytopia) [reply check_http_expect:157]: Looks like you're absolutely right. That's something I didn't catch in the review and also didn't read the error message properly. :-)

**Review by cytopia — APPROVED** [review]

**Pull request merged**